examples/ex10/main.cpp

```diff
@@ -0,0 +1,429 @@
+
+#include <iostream>
+#include <cmath>
+#include <numeric>
+#include <algorithm>
+
+#include "FeedForwardNeuralNetwork.hpp"
+#include "PrintUtilities.hpp"
+
+#include <gsl/gsl_rng.h>
+#include <gsl/gsl_randist.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_vector.h>
+#include <gsl/gsl_blas.h>
+#include <gsl/gsl_multifit_nlinear.h>
+
+
+/*
+Example: Train FFNN via GSL's Non-Linear Fit with Levenberg-Marquardt solver
+
+In this example we want to demonstrate how a FFNN can be trained via GSL's fit routines to fit target data (in this case a gaussian).
+
+The fit is achieved by minimizing the mean square distance of NN to data (by Levenberg-Marquardt), where the distance and gradient are integrated on a unifrom grid.
+Hence, we have to create the NN and target data, define functions for distance and gradient, and pass all the information to the GSL lib.
+
+For convenience, a NNFitter1D class is created, which handles the whole process in a user-friendly way, including normalization of the data.
+*/
+
+
+double Gaussian(const double &x, const double &a, const double &b) {
+  return exp(-a*pow(x-b, 2));
+};
+
+struct data {
```

Comment:
data is quite a generic name... wither give a more specific name or explain what is this for in a comment